src/tasks/addonindex/taskaddonindex.cpp

```diff
@@ -48,23 +52,27 @@ void TaskAddonIndex::run() {
     connect(request, &NetworkRequest::requestFailed, this,
             [this](QNetworkReply::NetworkError error, const QByteArray&) {
               logger.error() << "Get addon index signature failed" << error;
-              emit completed();
+
+              m_indexSignatureRequestComplete = true;
+              maybeComplete();
             });
 
     connect(request, &NetworkRequest::requestCompleted, this,
             [this](const QByteArray& data) {
               logger.debug() << "Get addon index signature completed";
+
               m_indexSignData = data;
+              m_indexSignatureRequestComplete = true;
               maybeComplete();
             });
   }
 }
 
 void TaskAddonIndex::maybeComplete() {
-  if (m_indexData.isEmpty() || m_indexSignData.isEmpty()) {
+  if (!m_indexSignatureRequestComplete || !m_indexRequestComplete) {
```

Comment:
I would like to suggest a different approach. If the signature verification is not needed, we could skip the second network-request.
And in here, we can check the feature again. If the signature verification is not requested, we can ignore the indexSignatureRequestComplete.

Reply:
Yeah that was my first approach here, but I was worried about adding signature checks in places outside of the AddonIndex module and then if we ever change it, forget to change it here. That is something that should be mitigated by having a test that fails in case we forget to change it, now that I think about it... I'll change it.